Library/Formula/agda.rb

```diff
@@ -51,38 +56,81 @@ def install
     cabal_clean_lib
 
     if build.with? "stdlib"
-      resource("stdlib").stage prefix/"agda-stdlib"
+      resource("stdlib").stage prefix/"lib"
 
-      # install the standard library's helper tools
-      cd prefix/"agda-stdlib" do
+      # compile the standard library's helper tools
+      cd prefix/"lib" do
         cabal_sandbox do
           cabal_install "--only-dependencies"
-          cabal_install "--prefix=#{prefix/"agda-stdlib"}"
-          system prefix/"agda-stdlib"/"bin"/"GenerateEverything"
-        end
-        cabal_clean_lib
-      end
-
-      # install the standard library's FFI bindings for the MAlonzo backend
-      if build.with? "malonzo-ffi"
-        cd prefix/"agda-stdlib"/"ffi" do
-          cabal_install "--user"
+          cabal_install
+          system "GenerateEverything"
         end
       end
 
       # generate the standard library's documentation and vim highlighting files
-      cd prefix/"agda-stdlib" do
+      cd prefix/"lib" do
         system bin/"agda", "-i", ".", "-i", "src", "--html", "--vim", "README.agda"
       end
     end
 
-    # byte-compile and install Agda's included emacs mode
+    # compile the included Emacs mode
     if build.with? "emacs"
-      system bin/"agda-mode", "setup"
       system bin/"agda-mode", "compile"
     end
   end
 
+  def caveats
+    s = ""
+
+    if build.with? "stdlib"
+      s += <<-EOS.undent
+
+      The Agda standard library has been installed at:
+        #{lib}
+
+      To typecheck a program using the standard library, run:
+        agda -i . -i #{lib}/src Program.agda
+      EOS
+
+      if build.with? "malonzo"
+        s += <<-EOS.undent
+
+        To compile a program using the MAlonzo backend and the standard library's FFI bindings, run:
+          agda -i . -i #{lib}/src --ghc-flag=-i#{lib}/ffi -c Program.agda
+        EOS
+      end
+    end
+
+    if build.with? "emacs"
+      s += <<-EOS.undent
+
+      The Agda Emacs mode has been installed.  To add it to your ~/.emacs file, run:
+        agda-mode setup
+      EOS
+
+      if build.with? "stdlib"
+        s += <<-EOS.undent
+
+        The Emacs mode needs to be configured to use the standard library.  In Emacs, type:
+          M-x customize-group RET agda2 RET
+
+        Add the following path to Agda2 Include Dirs:
+          #{lib}/src
+        EOS
+
+        if build.with? "malonzo"
+          s += <<-EOS.undent
+
+          Add the following option to Agda2 Program Args:
+            --ghc-flag=-i#{lib}/ffi
+          EOS
+        end
+      end
+    end
+
+    s
+  end
```

Comment:
We don't want as many caveats as this. Strip it down to the bare minimum that is Homebrew-specific, thanks.